app/helpers/locations_helper.rb

```diff
@@ -42,12 +42,43 @@ def display_location_attributes(location)
     base_attributes
   end
 
+  ActivePeriod = Struct.new(:from, :to)
+  def active_periods(location)
+    periods = [ActivePeriod.new(location.available_from, nil)]
+
+    sorted_deactivation_periods = remove_nested_periods(location.location_deactivation_periods.sort_by(&:deactivation_date))
+    sorted_deactivation_periods.each do |deactivation|
+      periods.find { |period| period.to.nil? }.to = deactivation.deactivation_date
+      periods << ActivePeriod.new(deactivation.reactivation_date, nil)
+    end
+
+    remove_overlapping_and_empty_periods(periods)
+  end
```

Comment:
```suggestion
  ActivePeriod = Struct.new(:from, :to)
  def active_periods(location)
    periods = [ActivePeriod.new(location.available_from, nil)]

    sorted_deactivation_periods = remove_nested_periods(location.location_deactivation_periods.sort_by(&:deactivation_date))
    sorted_deactivation_periods.each do |deactivation|
      periods.last.to = deactivation.deactivation_date
      periods << ActivePeriod.new(deactivation.reactivation_date, nil)
    end

    remove_overlapping_and_empty_periods(periods)
  end
```

I wonder: do we need to check whether the last period `to` is nil?

Here's another way to write it without structs. Whether this is easier to read I'm not convinced!

```suggestion
  def active_periods(location)
    periods = [[location.available_from, nil]]

    sorted_deactivation_periods = remove_nested_periods(location.location_deactivation_periods.sort_by(&:deactivation_date))
    sorted_deactivation_periods.each do |deactivation|
      periods.last[1] = deactivation.deactivation_date
      periods << [deactivation.reactivation_date, nil]
    end

    remove_overlapping_and_empty_periods(periods)
  end
```

Reply:
Updated the .last suggestion, but I quite like the struct if there are no strong feelings there 👀

Reply:
None!